source/_posts/2019-01-28-building-a-decentralized-reddit-with-embark-part-2.md

```diff
@@ -0,0 +1,194 @@
+
+### Testing `createPost()`
+
+To test our Smart Contract's `createPost()` method, we'll make use of `DReddit`, which will now be a Smart Contract instance. If you remember, `createPost()` actually takes the post's description as bytes, so how do we make that work? Well, it turns out that we actually don't pass it the description itself, but an **IPFS hash** that points to the actual description. The reason for that is that posts can be very long, resulting in a lot of bytes. It's better to store the actual description in a storage where data size isn't an issue, and instead store a reference to that data in our Smart Contract. Using a hash makes the data size deterministic as it will always have the same length.
+
+Once we have such a hash (no worries, we've got one prepared), we can use Web3's `fromAscii()` utils to convert that hash to bytes and then send it off using our Smart Contract's `createPost()` method. We can then subscribe to the events we're emitting and check its return value like this:
+
+<pre class="javascript"><code>...
+const ipfsHash = 'Qmc5gCcjYypU7y28oCALwfSvxCBskLuPKWpK4qpterKC7z';
+
+contract('DReddit', () => {
+  ...
+  it ('should be able to create a post and receive it via contract event', async () => {
+    let receipt = await DReddit.methods.createPost(web3.utils.fromAscii(ipfsHash)).send();
+    const event = receipt.events.NewPost;
+    postId = event.returnValues.postId;
+    assert.equal(web3.utils.toAscii(event.returnValues.description), ipfsHash);
+  });
+});</code></pre>
+
+Notice that we're using `async/await` here because Embark's Smart Contract instance methods return promises. The same can be done without promises as well, it's just a syntactical difference at this point. Running `embark test` should result in two passing tests now!
+
+## Testing correctness of data
+
+Another good test case would be to check if the stored data such as the description bytes, the owner etc.  resolve back to the correct data. Notice that this is slightly different from what we're testing in our previous test - there we're testing the description bytes emitted by the `NewPost` event. To test this we take advantage of the `postId` created in the previous test, which is available globally now, to fetch the stored post. We then perform a similar check as in the previous test. We also want to test if the owner data of the post is correct, but for that we need to get access to the account that created the post in the first place.
```

Comment:
Not sure, it feels weird
```the owner etc.  resolve ```
=>
```the owner etc...  resolve ```

Reply:
The way Pascal wrote it is correct as far as I know